Covinator/Covinator/Pages/PfizerVaccination.cshtml.cs

```diff
@@ -29,27 +29,32 @@ public void OnGet()
                 JSchema schema = JSchema.Parse(System.IO.File.ReadAllText("pfizerSchema.json"));
                 JArray jsonArray = JArray.Parse(pfizerData);
                 IList<string> validationEvents = new List<string>();
-                var currentWeek1 = jsonArray[0].ToArray();
-                var currentWeek = currentWeek1[1].ToString();
+                var currentWeek1 = jsonArray[0];
+                
+                var currentWeek = currentWeek1["week_of_allocations"];
+
+
                 if (jsonArray.IsValid(schema, out validationEvents))
                 {
 
                     JArray result_array = new JArray();
                     JArray jarray = new JArray();
                     var length_arr = jsonArray.Count();
-                    for (int i=0; i<63; i++)
+                    
+            
+                    foreach (JObject i in jsonArray
+                        .Where(obj => obj["week_of_allocations"].ToString() == currentWeek.ToString()))
```

Comment:
•	On Pfizer page, the table for Vaccine distribution by states for the current week had hardcoding for fetching the number of elements on OnGet. I used the above logic to get all the elements for the latest week.

Source: https://stackoverflow.com/questions/10543512/how-do-i-enumerate-through-a-jobject